### DISCUSSION TITLE
How do I update a Github Action that I have previously published?

### DISCUSSION THREAD

**Discussion by richb-hanover** [community/community#129097] [Actions] [answered]
### Select Topic Area

Question

### Body

I published a Github Action (richb-hanover/cargo). It works fine, except that it relies on Node.js 16, which is deprecated.

I have updated the _action.yml_ file to invoke Node20. But people are still getting the "Node 16 is deprecated" message.

How do I cause the new file to be published? Is there a clear description somewhere? (I've been asked the same question by someone else, so I'm looking for an answer for two people...) Many thanks.

**Comment by Rishi-Sudhakar** [comment]
## Update the `action.yml` File

Ensure that your `action.yml` file specifies Node.js 20. This usually involves updating the `runs` section.

```yaml
runs:
  using: 'node20'

**Reply by richb-hanover** [reply]
Yes - I have done this. https://github.com/richb-hanover/cargo/blob/master/action.yml

Do I need to do anything to make the new version active? (My google-fu isn't strong enough to find a "trigger" for publishing/republishing/updating an action...) Thanks

**Reply by Rishi-Sudhakar** [reply, marked as answer]
It automatically does CI/CD(you don't have to manually run), I saw your actions file. Seems the deployment has failed at build. This issue can arise when using cryptographic operations that are not compatible with OpenSSL 3.0, which Node.js 17+ uses by default. So you're struck at 16, if you use 16+ the build will fail.

**Reply by RasoulAzwd01** [reply]
😎 cool

**Comment by richb-hanover** [comment]
Yes - I understand what you're saying. (When I look at Actions tab, I can see that _.github/workflows/ci.yml_ attempts to build the file and it has trouble.)

Fortunately, I found another package that does this work and is maintained (clechasseur/rs-cargo@v2), so I don't have to fix my action/repo. Thanks!

**Comment by RasoulAzwd01** [comment]
> ### Select Topic Area
> 
> Question
> 
> ### Body
> 
> I published a Github Action (richb-hanover/cargo). It works fine, except that it relies on Node.js 16, which is deprecated.
> 
> I have updated the _action.yml_ file to invoke Node20. But people are still getting the "Node 16 is deprecated" message.
> 
> How do I cause the new file to be published? Is there a clear description somewhere? (I've been asked the same question by someone else, so I'm looking for an answer for two people...) Many thanks.

Ok

**Comment by RasoulAzwd01** [comment]
Best work about this question ⁉️

**Comment by Inonameraja** [comment]
runs:
  using: 'node20'

**Comment by Inonameraja** [comment]
jobs:
  jobName:
    runs-on: ubuntu-24.04

**Comment by Inonameraja** [comment]
https://github.com/privacysandbox/privacy-sandbox-dev-support.git

**Comment by Inonameraja** [comment]
allow hal_module_label secure_element_service:service_manager find

**Comment by Inonameraja** [comment]
document.cookie="__Host-name=value; Secure; Path=/; SameSite=None; Partitioned;"
